src/rest_api/api/indexing/bulk_index.rs

```diff
@@ -133,11 +133,11 @@ fn bulk_request_doc_to_tantivy_doc(
 
     let mut doc = tantivy::doc!(
         smiles_field => tautomer.as_smiles(),
-        fingerprint_field => fingerprint.0.into_vec()
+        fingerprint_field => fingerprint.0.clone().into_vec()
```

Comment:
I'm usually suspicious of `clone()` because of the precious memory. Could the `doc!` macro take a slice of bytes instead of a vec of bytes?

```
    let mut doc = tantivy::doc!(
        smiles_field => tautomer.as_smiles(),
        fingerprint_field => &fingerprint.0[..]
        ...
```

or does it error? I bet it does but it might be worth a little look. This is the `doc!` macro:

```
macro_rules! doc(
    () => {
        {
            ($crate::TantivyDocument::default())
        }
    }; // avoids a warning due to the useless `mut`.
    ($($field:expr => $value:expr),*) => {
        {
            let mut document = $crate::TantivyDocument::default();
            $(
                document.add_field_value($field, $value);
            )*
            document
        }
    };
    // if there is a trailing comma retry with the trailing comma stripped.
    ($($field:expr => $value:expr),+ ,) => {
        doc!( $( $field => $value ), *)
    };
);
```

which hinges on `add_field_values` whose type looks like:

```
pub fn add_field_value<T: Into<OwnedValue>>(&mut self, field: Field, typed_val: T)
```

so maybe that won't work! but give it a shot? I just want to make sure we are deffo questioning `clone()` if possible.

Reply:
Where there's a will, there's a way...

Seem's like `.as_raw_slice()` gets the job done without cloning :)